### ISSUE TITLE
UTC vs Floating Time not represented

### ISSUE THREAD

**Issue by ocelotsloth** [sebbo2002/ical-generator#52]
## tl;dr The Problem
Currently, UTC time is the assumption when it should instead be floating time.

**Important Note:** This is at the _event_ level, not the calendar level.

## Example/Details
To explain let's look at the `example_toString.js` file and how it works/behaves:

### The JavaScript / Intent
This is the relevant code in the example described above:
```javascript
event = cal.createEvent({
    start: new Date(new Date().getTime() + 3600000),
    end: new Date(new Date().getTime() + 7200000),
    summary: 'Example Event',
    description: 'It works ;)',
    organizer: 'Organizer\'s Name <organizer@example.com>',
    url: 'http://sebbo.net/'
});
```
Specifically, the `start` and `end` is created with a `new Date(new Date.getTime() + delay)` call. These calls are done from the timezone of the computer running the code be it on a server or a user's machine. 

It's important to note that this example has _not_ defined a timezone for the event.

Because the code has not yet specified _any_ particular timezone (be it -5 or UTC), the iCal generator should be operating in "floating" timezone mode. Looking at RFC 5545, here is where "floating" mode is defined (3.3.5):
```
FORM #1: DATE WITH LOCAL TIME

      The date with local time form is simply a DATE-TIME value that
      does not contain the UTC designator nor does it reference a time
      zone.  For example, the following represents January 18, 1998, at
      11 PM:

       19980118T230000

      DATE-TIME values of this type are said to be "floating" and are
      not bound to any time zone in particular.  They are used to
      represent the same hour, minute, and second value regardless of
      which time zone is currently being observed.
```

Basically, if we give 2pm as a time, it should store _just_ the 2pm, and not any timezone data at all. This is not going to be the behavior that most want out of the calendar, for the very reasons we have timezones in the first place.

That said, here is what the current library generates:
```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//sebbo.net//ical-generator//EN
BEGIN:VEVENT
UID:e22s@example.com
SEQUENCE:0
DTSTAMP:20170110T023408Z
DTSTART:20170110T033408Z
DTEND:20170110T043408Z
SUMMARY:Example Event
DESCRIPTION:It still works \;)
ORGANIZER;CN="Organizer's Name":mailto:organizer@example.com
URL;VALUE=URI:http://sebbo.net/
END:VEVENT
END:VCALENDAR
```
Paying special attention to the dates, we can see that the start time is defined as `20170110T023408Z`. This is in fact the correct time (I ran it at around 10pm EST). This causes some unique behavior from programs like MS Outlook, which gives a warning when it does the UTC conversion, which may not be what somebody wants in their end product.

This format is instead in the format of UTC, as defined in the RFC (3.3.5) as follows:
```
FORM #2: DATE WITH UTC TIME

      The date with UTC time, or absolute time, is identified by a LATIN
      CAPITAL LETTER Z suffix character, the UTC designator, appended to
      the time value.  For example, the following represents January 19,
      1998, at 0700 UTC:

       19980119T070000Z

      The "TZID" property parameter MUST NOT be applied to DATE-TIME
      properties whose time values are specified in UTC.
```

## The Solution / Proposed Fix

I would like to see an option to support both timezones and UTC. I imagine that this could be done by assuming floating when the timezone is `null` and a new boolean value for UTC is set to `false`.

In such a system, I imagine that the library would need to throw an exception when UTC is set to `true` and the timezone is not `null`. An alternative would be to simply ignore UTC if there is a timezone set and not throw an exception.

**Comment by ocelotsloth** [comment]
Upon further inspection, I think I found the bug, looking at `event.js` and specifically the `generate` method:

https://github.com/sebbo2002/ical-generator/blob/1b3bc74d90f5365756a0db765e35af6275b6e088/lib/event.js#L689-L693
```javascript
        if(data.allDay) {
            g += 'DTSTART;VALUE=DATE:' + tools.formatDate(data.start, true) + '\r\n';
            if(data.end) {
                g += 'DTEND;VALUE=DATE:' + tools.formatDate(data.end, true) + '\r\n';
}
```
At the function call to `tools.formatDate(data.start, true)` shouldn't that `true` be instead the value of `data.floating`?

**Comment by ocelotsloth** [comment]
After fixing that issue, I found an additional problem. The current implementation still uses the UTC time instead of local when generating the time strings, just doesn't add the Z onto the end.

```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//sebbo.net//ical-generator//EN
BEGIN:VEVENT
UID:ab1z@example.com
SEQUENCE:0
DTSTAMP:20170110T032656Z
DTSTART:20170110T032656
DTEND:20170110T052656
SUMMARY:Example Event
DESCRIPTION:It still works \;)
ORGANIZER;CN="Organizer's Name":mailto:organizer@example.com
URL;VALUE=URI:http://sebbo.net/
END:VEVENT
END:VCALENDAR
```

from:
```javascript
event = cal.createEvent({
    start: new Date(new Date().getTime()),
    end: new Date(new Date().getTime() + 7200000),
    floating: true,
    summary: 'Example Event',
    description: 'It works ;)',
    organizer: 'Organizer\'s Name <organizer@example.com>',
    url: 'http://sebbo.net/'
});
```

**Comment by sebbo2002** [comment]
As I noted [here](https://github.com/sebbo2002/ical-generator/pull/53#issuecomment-271904291), I think it's better to wait till I build the 3.0, as that's a minor change and there are other breaking topics on my list as well.

**Comment by sebbo2002** [comment]
With the current develop, timestamps should be output correctly via the local timestamp or via UTC. Can the issue be closed or is there still something to do here and if so, what?

**Comment by sebbo2002** [comment]
Closing due to inactivity. Please reopen in case this is not fixed…
